background/lib/four-byte.ts

```diff
@@ -0,0 +1,84 @@
+import { fetchJson } from "@ethersproject/web"
+import { Interface } from "@ethersproject/abi"
+import { HexString } from "../types"
+import logger from "./logger"
+
+export type FourByteSignature = {
+  id: string
+  createdAt: string
+  functionSignature: string
+  functionSelector: string
+}
+
+/**
+ * Calculate a function selector from an ABI fragment.
+ *
+ * @param functionSignature an ABI fragment used to get the function selector,
+ *        in the form `"function(uint256,uint256)"`
+ */
+export function calculateFunctionSelector(functionSignature: string): string {
+  const abi = [`function ${functionSignature}`]
+  const iface = new Interface(abi)
+  return iface.getSighash(Object.keys(iface.functions)[0])
+}
```

Comment:
```suggestion
export function calculateFunctionSelector(functionSignature: string): string {
  return FunctionFragment.from(`function ${functionSignature}`).format("sighash")
}
```
Believe this will do the trick, and may honestly invalidate the need for the wrapper function.